components/common/buttons/contract-button/contract-button.view.tsx

```diff
@@ -0,0 +1,39 @@
+import Button from "components/button";
+import type {ContractButtonProps} from "components/common/buttons/contract-button/contract-button.controller";
+import WalletMismatchModal from "components/modals/wallet-mismatch/controller";
+import WrongNetworkModal from "components/wrong-network-modal";
+
+interface ContractButtonViewProps extends ContractButtonProps {
+  isNetworkModalVisible?: boolean;
+  isMismatchModalVisible?: boolean;
+  isRequired?: boolean;
+  onCloseModal: (modal: "WalletMismatchModal" | "WrongNetworkModal") => void;
+}
+export default function ContractButtonView({
+  isNetworkModalVisible,
+  isMismatchModalVisible,
+  isRequired,
+  onCloseModal,
+  children,
+  ...rest
+}: ContractButtonViewProps) {
+  return(
+    <>
+      <Button
+        {...rest}
+      >
+        {children}
+      </Button>
+
+      <WrongNetworkModal
```

Comment:
this will create the same amount of "wrong network modals" as the numbers of "contract-buttons" on the page -- we should not do this, we should have this at the top of the page and then show/hide depending on the state; this to avoid having more listeners than the ones needed to inform the user about X or Y